pkg/controller/handlers/knowledgefile/knowledgefile.go

```diff
@@ -46,7 +46,8 @@ func shouldReIngest(file *v1.KnowledgeFile) bool {
 	return file.Spec.IngestGeneration > file.Status.IngestGeneration ||
 		file.Spec.UpdatedAt != file.Status.UpdatedAt ||
 		file.Spec.Checksum != file.Status.Checksum ||
-		file.Spec.URL != file.Status.URL
+		file.Spec.URL != file.Status.URL ||
+		(file.Status.State == types.KnowledgeFileStateError && file.Status.RetryCount < 3)
```

Comment:
Should we parameterize the number of retries? Maybe even give it an undocumented env for testing?

Reply:
I think let's not complicate this for now. It should be enough to resolve most transient error and have proper retry.